event-publisher-spi/src/main/scala/uk/gov/nationalarchives/eventpublisherspi/EventPublisherProviderFactory.scala

```diff
@@ -21,7 +23,16 @@ class EventPublisherProviderFactory extends EventListenerProviderFactory {
     eventPublisherConfig = Option(EventPublisherConfig(snsUrl, snsTopicArn, tdrEnvironment))
   }
 
-  override def postInit(factory: KeycloakSessionFactory): Unit = { }
+  override def postInit(factory: KeycloakSessionFactory): Unit = {
+    factory.register(event => {
+      if(event.isInstanceOf[PostMigrationEvent]) {
+         val session = factory.create()
+        val provider: TimerProvider = session.getProvider(classOf[TimerProvider])
+        val task = UserMonitoringTask(eventPublisherConfig.get)
+        provider.scheduleTask(task, 24 * 60 * 60 * 1000, "test")
```

Comment:
Should the `taskName` be `test`? Is it worth pulling out the timer into a val, and giving it a name to make it clearer that the unit is milliseconds?

Reply:
I didn't mean to leave it as test :smile: I've renamed it and moved the interval out to a variable.